src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,91 @@
+package duke;
+
+import java.util.Scanner;
+
+/** Handles input/output for Duke. */
+public class Ui {
+
+    private Scanner scanner;
+    private StringBuilder msg;
+    private boolean doPromptUserInput;
```

Comment:
Not sure if this helps, maybe naming it as `toPromptUserInput` could be clearer?

Reply:
actually on second thought maybe it should be `shouldPromptUserInput` 🤔